### PR TITLE
[blog] Add Go SDK blog

### PR DESCRIPTION
**Pull request by ericgregory**
Go SDK blog WIP. Needs final result before merging.

### CODE REVIEW

**Comment by ericgregory** [comment]
This is still blocked by the error at the end, *but* when it’s unblocked we should also make sure that [this PR](https://github.com/wasmCloud/wasmCloud/pull/2349) has merged first for the instructions in the example to be correct.

File: blog/go-provider-sdk/index.mdx [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	// Initialize the provider with callbacks to track linked components
+	providerHandler := Handler{}
```
Comment (brooksmtownsend) [inline blog/go-provider-sdk/index.mdx]: Just needs update from https://github.com/wasmCloud/wasmCloud/pull/2388

File: blog/go-provider-sdk/index.mdx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+func handleHealthCheck(handler *Handler) string {
+	handler.provider.Logger.Info("Handling health check")
+	return "provider healthy"
+}
+
+func handleShutdown(handler *Handler) error {
+	handler.provider.Logger.Info("Handling shutdown")
+	clear(handler.linkedFrom)
+	clear(handler.linkedTo)
+	return nil
```
Comment (brooksmtownsend) [inline blog/go-provider-sdk/index.mdx]: These just need the println fix from https://github.com/wasmCloud/wasmCloud/pull/2386

**Review by brooksmtownsend — APPROVED** [review]

**Review by brooksmtownsend — APPROVED** [review]

**Comment by ericgregory** [comment]
Dated for July 2, ready to merge then.

**Review by CazJT — APPROVED** [review]

**Pull request merged**